component/file_cache/file_cache.go

```diff
@@ -691,9 +691,44 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) getHandleData(handle *handlemap.Handle) *handlemap.Handle {
+
+	log.Debug("FileCache::getHandleData : Need to download %s", handle.Path)
+
+	//exctract the flags out of handle values
+	flags, _ := handle.GetValue("flag")
+	flagsStruct, ok := flags.(struct{ flags int })
+	if !ok {
+		log.Debug("FileCache::getHandleData : error Type assertion failed on getting flag for %s [%s]", handle.Path)
+		return nil
+	}
+
+	// Extract the integer as a separate variable
+	flag := flagsStruct.flags
+
+	// extract the filemode out of handle values
+	mode, _ := handle.GetValue("mode")
+	fileModeValue, ok := mode.(os.FileMode)
+	if !ok {
+		log.Debug("FileCache::getHandleData : error Type assertion failed on getting file mode for %s [%s]", handle.Path)
+		return nil
+	}
+
+	handle, err := fc.DownloadFile(internal.DownloadFileOptions{Name: handle.Path, Flags: flag, Mode: fileModeValue, Handle: handle})
```

Comment:
getHandleData is more about fetching the handle data that's initially missing. It goes and downloads and loads in the rest of the handle data that wasn't there previously. The flag and mode is a kind of assurance to make sure the correct permissions and filemode is used while the file being used for the download has the correct configuration.

Reply:
Ok, now I get why it's called that.
I do feel like `getHandleData` is a bit vague, compared to `DonwloadFile`, which is direct and clear.
But anyway, this comment is just about simplifying and improving readability, which is less important than the concurrency and error handling issues above.

Reply:
I'm open to better names if anyone has a suggestion. I'll give it some thought. downloadHandleData seems a bit long.

Reply:
Since the getHandleData is so small, It would be easy to incorperate the same code into DownloadFile() and get rid of getHandleData. I'll go ahead and apply what you're suggesting. Just like bad jokes, if it needs explaining, then it needs work.